src/main/java/dinkplugin/notifiers/GambleNotifier.java

```diff
@@ -0,0 +1,128 @@
+package dinkplugin.notifiers;
+
+import com.google.common.collect.ImmutableSet;
+import dinkplugin.message.NotificationBody;
+import dinkplugin.message.NotificationType;
+import dinkplugin.notifiers.data.GambleNotificationData;
+import dinkplugin.notifiers.data.SerializedItemStack;
+import dinkplugin.util.ItemSearcher;
+import dinkplugin.util.ItemUtils;
+import dinkplugin.util.Utils;
+import lombok.NonNull;
+import lombok.Value;
+import lombok.extern.slf4j.Slf4j;
+import net.runelite.client.game.ItemManager;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.tuple.Pair;
+import org.jetbrains.annotations.Nullable;
+import org.jetbrains.annotations.VisibleForTesting;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.util.Collection;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@Slf4j
+@Singleton
+public class GambleNotifier extends BaseNotifier {
+    // itemName (x quantity)![ tertiaryReward!] gc: n
+    private static final Pattern GAMBLE_REGEX = Pattern.compile("^(.+?)!\\s*(.+!)?\\s+High level gamble count: (\\d+).*");
+    private static final Pattern ITEM_QUANTITY_REGEX = Pattern.compile("^(.+?)\\(x (\\d+)\\)$");
+    private static final Collection<String> RARE_LOOT = ImmutableSet.of("dragon chainbody", "dragon med helm");
```

Comment:
Is [pet penance queen](https://oldschool.runescape.wiki/w/Barbarian_Assault/Rewards#Tertiary_High_Gamble_Rewards) covered by this approach?

Reply:
Unfortunately, the pet doesn't show up anywhere in the message upon receiving it. I learned this yesterday from experience :frowning_face:

Reply:
Did you receive the normal `You have a funny feeling...` chat message? If so then it will just be handled by our existing pet notifier.

Reply:
Yeah, the normal pet drop game message appears, so it will be handled by the pet notifier

Reply:
> Unfortunately, the pet doesn't show up anywhere in the message upon receiving it. I learned this yesterday from experience ☹️

Ah that's unfortunate - perhaps we can add it to the rare set anyways (in case jagex changes this) with a comment explaining it currently doesn't occur